network/peer/peer.go

```diff
@@ -702,7 +706,114 @@ func (p *peer) handle(msg message.InboundMessage) {
 	p.Router.HandleInbound(context.Background(), msg)
 }
 
-func (p *peer) handlePing(*p2p.Ping) {
+func (p *peer) handlePing(msg *p2p.Ping) {
+	// if GetUptime is not 0 then this is a guaranteed new ping message
+	// It can be 0 in both cases: 1) old ping message
+	// 2) new ping message and we're not a validator (or an actual 0 uptime)
+	// In both cases we should handle it as an old ping message since
+	// it is backwards compatible
+	// TODO: remove this in the future
+	if msg.GetUptime() != 0 {
+		p.handleNewPing(msg)
+	} else {
+		p.handleOldPing(msg)
+	}
+}
+
+func (p *peer) createPingMessage() (message.OutboundMessage, error) {
```

Comment:
Can we share this code with `handleOldPing`? They are doing almost exactly the same thing

Reply:
done